.github/workflows/ci-baseline.yml

```diff
@@ -28,75 +28,92 @@ jobs:
+          - ubuntu-20.04-clang_9-eigen_3.3.7-gtest
+          - fedora-34-gnu_11-eigen_3.3.7-gtest
+          - fedora-34-clang_12-eigen_3.3.7-gtest
+          #- intel_oneapi-eigen_3.3.7-gtest
+        build_type:
+          - Release
+          - Debug
+        cmake_extra_args:
+          - ''
+        include:
+          - image: ubuntu-20.04_gnu-11_eigen-3.3.7_gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
+          - image: ubuntu-20.04_clang-12_eigen-3.3.7_gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
+          - image: fedora-34-gnu_11-eigen_3.3.7-gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
+          - image: fedora-34-clang_12-eigen_3.3.7-gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
 
-    runs-on: ${{ matrix.config.os }}
-    container: pressiomodelreduction/${{ matrix.config.image }}
+    runs-on: ubuntu-latest
+    container: pressiomodelreduction/${{ matrix.image }}
     env:
-      num_cpus: 2
+      num_cpus: 2 # $(cat /proc/cpuinfo | grep processor -c)
       eigen_inc_dir: /usr/local/eigen/install/include/eigen3
       gtest_dir: /usr/local/gtest/install
-      small: small
-      medium: medium
-      large: large
 
     steps:
       - uses: actions/checkout@v3
 
-      - name: configure
+      - name: Configure
         run: |
           cmake -B builddir \
-            -D CMAKE_BUILD_TYPE:STRING=${{ matrix.config.mode }} \
-            -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+            -D CMAKE_BUILD_TYPE:STRING=${{ matrix.build_type }} \
             -D CMAKE_C_COMPILER=$CC \
             -D CMAKE_CXX_COMPILER=$CXX \
-            -D PRESSIO_ENABLE_TPL_EIGEN:BOOL=ON \
-            -D EIGEN_INCLUDE_DIR=$eigen_inc_dir \
+            -D CMAKE_CXX_FLAGS='' \
+            -D CMAKE_INSTALL_PREFIX:PATH=../install \
+            -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+            ${{ matrix.cmake_extra_args }} \
             -D GTEST_ROOT=$gtest_dir \
             -D PRESSIO_ENABLE_DEBUG_PRINT=ON \
             -D PRESSIO_ENABLE_TESTS:BOOL=ON \
-            -D CMAKE_INSTALL_PREFIX:PATH=../install \
-            -D CMAKE_CXX_FLAGS=''
+            -D PRESSIO_ENABLE_TPL_EIGEN:BOOL=ON \
+            -D EIGEN_INCLUDE_DIR=$eigen_inc_dir
 
-      # In this stage small functional tests are built and run
-      - name: build-test-${{ env.small }}
+      - name: Build
         run: |
-          cd builddir/tests/functional_${{ env.small }}
-          make -j $num_cpus
-          ctest -j $num_cpus --output-on-failure
+          cmake --build builddir --target install
 
-      # In this stage medium functional tests are built and run
-      - name: build-test-${{ env.medium }}
+      - name: Test
+        working-directory: builddir
```

Comment:
why are you getting rid of what was done before? we need to keep the small/medium/large separation. 
this was very important because if the small tests do not pass, there is no need to build/run the medium and large. 
so please keep this similar to what was done before. we ned the outputonfailure after small,medium and large.

Reply:
Note that `functional_medium` and `functional_large` directories are empty on `develop` - all the tests are in `functional_small` anyways.

Reply:
i know but they are empty for now...

Reply:
Added back the small/medium/large separation. Note that there is only one build stage now and only running the tests is separated.
This is because we need to build `install` target in order to build the examples (which are run before the tests). We can split it further and build the tests separately if needed.